src/js/PhotoSphereViewer.core.js

```diff
@@ -163,13 +174,101 @@ PhotoSphereViewer.prototype._loadTexture = function() {
       throw new PSVError('Cannot load image');
     };
 
-    loader.load(self.config.panorama, onload, onprogress, onerror);
+    if (true === self.prop.cacheTextures && 'undefined' != typeof self.prop.loadedTextures[self.config.panorama]) {
+      defer.resolve(self.prop.loadedTextures[self.config.panorama]);
+    }
+    else {
+      if (true === self.prop.cacheTextures) {
+        self.prop._loadingTextures.push(self.config.panorama);
+      }
+      loader.load(self.config.panorama, onload, onprogress, onerror);
+    }
 
     return defer.promise;
   });
 };
 
 /**
+ * Loads the sphere texture
+ * @returns {promise}
+ * @private
+ */
+PhotoSphereViewer.prototype._preloadPanorama = function(pano, progressCallback) {
+  var self = this;
+  var defer = D();
+  var loader = new THREE.ImageLoader();
+  var progress = 0;
+
+  loader.setCrossOrigin('anonymous');
+
+  var onload = function(img) {
+    if (self.loader) {
```

Comment:
The idea is to show the marker in the view, but 'disabled'
And as soon as the panorama get loaded, the marker become 'enabled'